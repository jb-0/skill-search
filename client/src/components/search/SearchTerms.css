.search-term-container {
  background-color: white;
  display:inline-block;
  background-color: #fb8500;
  margin: 5px;
  padding: 0 10px;
  border-radius: 15px;
}