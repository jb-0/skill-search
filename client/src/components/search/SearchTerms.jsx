import React from 'react';
import './SearchTerms.css'

function SearchTerms(props) {
  return (
    <div>
      {props.searchTerms.map((searchTerm) => {
        return (
          <div className="search-term-container">
            <p onClick={props.removeSearchTerm} id={searchTerm}>
              x {searchTerm}
            </p>
          </div>
        );
      })}
    </div>
  );
}

export default SearchTerms;