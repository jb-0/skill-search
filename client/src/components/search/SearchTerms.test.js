import '@testing-library/jest-dom/extend-expect';
import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";

import SearchTerms from "./SearchTerms";

describe('SearchTerms component', () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('terms passed in via array are rendered successfully and prefixed with "x"', () => {
    act(() => {
      render(<SearchTerms searchTerms={['node', 'sql']} />, container);
    });

    expect(screen.getByText('x node')).toBeInTheDocument();
    expect(screen.getByText('x sql')).toBeInTheDocument();
  });

  it('when an empty array is passed, no terms (prefixed with "x") are rendered', () => {
    act(() => {
      render(<SearchTerms searchTerms={[]} />, container);
    });

    expect(screen.queryByText('x')).toBeNull();
  });

  it('renders correctly when passed an array containing search terms', () => {
    const tree = renderer
      .create(<SearchTerms searchTerms={['node', 'sql']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when passed an empty array', () => {
    const tree = renderer
      .create(<SearchTerms searchTerms={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });  
});
