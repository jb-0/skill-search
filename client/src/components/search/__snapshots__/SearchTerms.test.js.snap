// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTerms component renders correctly when passed an array containing search terms 1`] = `
<div>
  <div
    className="search-term-container"
    data-testid="search-term-container"
  >
    <p
      id="node"
    >
      x 
      node
    </p>
  </div>
  <div
    className="search-term-container"
    data-testid="search-term-container"
  >
    <p
      id="sql"
    >
      x 
      sql
    </p>
  </div>
</div>
`;

exports[`SearchTerms component renders correctly when passed an empty array 1`] = `<div />`;
